homeworkW11D02/src/ba/bitcamp/homeworkW11D02/Server.java

```diff
@@ -0,0 +1,234 @@
+		upPanel.setLayout(new BorderLayout());
+		upPanel.add(scrollPane, BorderLayout.CENTER);
+
+		/*
+		 * Configuring text area
+		 */
+		textArea.setLineWrap(true);
+		textArea.setEditable(false);
+		textArea.setForeground(Color.BLUE);
+		textArea.setBackground(Color.ORANGE);
+		textArea.setFont(new Font("Serif", Font.BOLD, 15));
+
+		/*
+		 * Configuring and adding components to main panel
+		 */
+		mainPanel.setLayout(new BorderLayout());
+		mainPanel.add(upPanel, BorderLayout.CENTER);
+		mainPanel.add(downPanel, BorderLayout.SOUTH);
+
+		// Adding main panel to the window
+		add(mainPanel);
+
+		/*
+		 * Configuring window
+		 */
+		setTitle("Server chat");
+		setSize(450, 600);
+		setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+		setResizable(false);
+		setLocationRelativeTo(null);
+		setVisible(true);
+
+		/*
+		 * Starting server
+		 */
+		try {
+			server = new ServerSocket(6699);
+
+			// Waiting for client to connect
+			client = server.accept();
+
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+		BufferedReader reader;
+		try {
+
+			/*
+			 * Reading messages from client
+			 */
+			reader = new BufferedReader(new InputStreamReader(
+					client.getInputStream()));
+			boolean end = false;
+			while (!end) {
+				String readText = "";
+				String line = reader.readLine();
+
+				/*
+				 * If message starts with "/list" followed with file path, it is
+				 * command to find all subdirectories in certain directory on
+				 * server's PC and put them in to a list. After that list shows
+				 * in both server's and client's chat room.
+				 */
+				if (line.indexOf("/list") != -1) {
+					listOfDirectories = new ArrayList<String>();
+					File currentDirectory = new File(line.substring(
+							line.indexOf("/list") + 6, line.length()));
+					if (currentDirectory.isDirectory()) {
+						File[] files = currentDirectory.listFiles();
+						for (File file : files) {
+							if (file.isDirectory()) {
+								listOfDirectories.add(file.getName());
+							}
+						}
+					}
+					textArea.append("Server: " + listOfDirectories.toString()
+							+ "\n");
+					BufferedWriter writer = new BufferedWriter(
+							new OutputStreamWriter(client.getOutputStream()));
+					writer.write("Server: " + listOfDirectories.toString()
+							+ "\n");
+					writer.flush();
+
+					/*
+					 * If message from client starts with "/web" followed with
+					 * web link, it is a command to open certain web page in
+					 * local browser on server's PC
+					 */
+				} else if (line.indexOf("/web") != -1) {
+					Desktop.getDesktop().browse(
+							new URI(line.substring(line.indexOf("/web") + 5,
+									line.length())));
+
+					/*
+					 * If message from client starts with "/delete" followed
+					 * with file path, it is a command to delete certain file
+					 * from server's PC
+					 */
+				} else if (line.indexOf("/delete") != -1) {
+					File file = new File(line.substring(
```

Comment:
Before deleting a file, one should check if `File` is a file or a directory and even if it exists or not.